### PR TITLE
Add deprecation warnings to libsox specific functions

### PR DESCRIPTION
**Pull request by mthrok**
*Note:* this should also go to master branch.

### CODE REVIEW

File: torchaudio/backend/sox_backend.py [hunk 1]
```diff
@@ -92,6 +92,9 @@ def info(filepath: str) -> Tuple[SignalInfo, EncodingInfo]:
 
 
 @_mod_utils.requires_module('torchaudio._torchaudio')
+@_mod_utils.deprecated(
+    'Please migrate to "sox_io" backend. See https://github.com/pytorch/audio/issues/903 for the detail',
+    '0.9.0')
 def save_encinfo(filepath: str,
```
Comment (vincentqb) [inline torchaudio/backend/sox_backend.py:98]: There is no mention of `save_encinfo` in #903. What is the user supposed to do?

File: torchaudio/backend/sox_backend.py [hunk 2]
```diff
@@ -224,6 +233,9 @@ def sox_encodinginfo_t() -> EncodingInfo:
 
 
 @_mod_utils.requires_module('torchaudio._torchaudio')
+@_mod_utils.deprecated(
+    'Please migrate to "sox_io" backend. See https://github.com/pytorch/audio/issues/903 for the detail',
+    '0.9.0')
```
Comment (vincentqb) [inline torchaudio/backend/sox_backend.py:238]: same, missing

File: torchaudio/backend/sox_backend.py [hunk 3]
```diff
@@ -163,6 +166,9 @@ def save_encinfo(filepath: str,
...
```
Comment (vincentqb) [inline torchaudio/backend/sox_backend.py:171]: The function is removed, and no substitute is provided, even in #903. What is a user supposed to do? I'd simply encourage them to user the new data format, here or in #903?

File: torchaudio/backend/sox_backend.py [hunk 4]
```diff
@@ -186,6 +192,9 @@ def sox_signalinfo_t() -> SignalInfo:
...
```
Comment (vincentqb) [inline torchaudio/backend/sox_backend.py:197]: same, instructions?

File: torchaudio/backend/sox_backend.py [hunk 5]
```diff
@@ -242,6 +254,9 @@ def get_sox_encoding_t(i: int = None) -> EncodingInfo:
...
```
Comment (vincentqb) [inline torchaudio/backend/sox_backend.py:259]: same, instructions?

File: torchaudio/backend/sox_backend.py [hunk 6]
```diff
@@ -259,6 +274,9 @@ def get_sox_option_t(i: int = 2) -> Any:
...
```
Comment (vincentqb) [inline torchaudio/backend/sox_backend.py:279]: same, instructions?

**Review by vincentqb — APPROVED** [review]
LGTM. I'd suggest adding a little bit of missing information to #903.

File: torchaudio/backend/sox_backend.py (hunk 3 above)
Comment (mthrok) [reply torchaudio/backend/sox_backend.py:171]: These are internal structures, which were not supposed to be exposed, therefore there is no replacement. Users can simply use the `save`/`load`/`info` functions. (therefore, `Please migrate to "sox_io" backend`).
I added the description in #903.

**Pull request merged**